keanu-python/keanu/algorithm/sampling.py

```diff
@@ -170,3 +188,50 @@ def _samples_generator(sample_iterator: JavaObject, vertices_unwrapped: JavaList
                 traces = []
 
         yield sample
+
+
+def __create_single_indexed_samples(network_samples: JavaObject, vertices_unwrapped: JavaList) -> Dict:
+    vertex_samples = {}
+    for vertex_unwrapped in vertices_unwrapped:
+        vertex_label = Vertex._get_python_label(vertex_unwrapped)
+        samples_for_vertex = network_samples.get(vertex_unwrapped).asList()
+        is_primitive = [True] * len(samples_for_vertex)
+        samples_as_ndarray = map(Tensor._to_ndarray, samples_for_vertex, is_primitive)
+        vertex_samples[vertex_label] = list(samples_as_ndarray)
+    return vertex_samples
+
+
+def __create_multi_indexed_samples(vertices_unwrapped: JavaList, network_samples: JavaObject, generated: bool) -> dict:
+    vertex_samples_multi: Dict = {}
+    tuple_hierarchy: Dict = {}
+    column_header_for_scalar = '(0)'
+    for vertex in vertices_unwrapped:
+        vertex_label = Vertex._get_python_label(vertex)
+        vertex_samples_multi[vertex_label] = defaultdict(list)
+        if generated:
+            sample = Tensor._to_ndarray(network_samples.get(vertex), return_as_primitive=True)
+            __add_sample_to_dict(sample, vertex_samples_multi, vertex_label, column_header_for_scalar)
+        else:
+            samples_for_vertex = network_samples.get(vertex).asList()
+            is_primitive = [True] * len(samples_for_vertex)
+            samples_as_ndarray = map(Tensor._to_ndarray, samples_for_vertex, is_primitive)
+            samples = list(samples_as_ndarray)
+            for sample in samples:
+                __add_sample_to_dict(sample, vertex_samples_multi, vertex_label, column_header_for_scalar)
+
+        tuple_hierarchy = {(vertex_label, tensor_index): values
+                           for vertex_label, tensor_index in vertex_samples_multi.items()
```

Comment:
should rename `tensor_index`, it's being used twice

Reply:
renamed